sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersClient.java

```diff
@@ -100,6 +129,21 @@ public PurchasedPhoneNumber getPurchasedPhoneNumber(String phoneNumber) {
         return asyncClient.beginSearchAvailablePhoneNumbers(countryCode, phoneNumberType, assignmentType, capabilities, searchOptions, context).getSyncPoller();
     }
 
+    /**
+     * Starts the purchase of the phone number(s) in the search result associated with a given id.
+     *
+     * This function returns a Long Running Operation poller that allows you to wait indefinitely until the
+     * operation is complete.
+     *
+     * @param searchId ID of the search
+     * @return A {@link SyncPoller} object.
```

Comment:
In line 101, the comment mentions what kind of SyncPoller is being returned `A {@link SyncPoller} object with the reservation result.` Do we need consistency?